portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,145 @@
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        ArrayList<Comment> comments = new ArrayList<>();
+        int index = 0;
+
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            if(index < numberOfShowedComments){
+                index++;
+            } else{
+                break;
+            }
+            String name = (String) entity.getProperty("name");
+            String content = (String) entity.getProperty("content");
+            comments.add(new Comment(name, content));
+        }
+
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        response.setContentType("application/json");
+        response.getWriter().println(json);
+    }
+
+    /**
+     * Obtain parameters of the HTML form and creates an entity in Datastore.
+     */
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String name = null;
+        String content = null;
+        boolean owo = false;
+        boolean uwu = false;
+
+        try {
+            name = getParameter(request, "name-input", "");
+            content = getParameter(request, "text-input", "");
+            owo = getBooleanParameter(request, "owo", false);
+            uwu = getBooleanParameter(request, "uwu", false);
+            DataServlet.numberOfShowedComments = getIntParameter(request, "num-input", 3);
+        } catch (NumberFormatException nfe) {
+            response.sendError(400, "Comment amount must be a number");
+            return;
+        } catch (Exception e){
+            response.sendError(500, "Error");
+            return;
+        }
+
+        String[] commentWords = content.split(" ", 0);
+        String commentConcatenated = "";
+        for(String word : commentWords){
+            if(owo){
+                word += "owo";
+            } else if (uwu){
+                word += "uwu";
+            } 
+            commentConcatenated += word + " ";
+        }
+
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("name", name);
+        commentEntity.setProperty("content", commentConcatenated);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+
+        response.sendRedirect("/index.html");
+    }
+
+    /**
+     * Helper method to convert an ArrayList of Comments to JSON format
+     * @param ArrayList<Comment> comments to be converted
+     * @return JSON as a String
+     */
+    private String convertToJsonUsingGson(ArrayList<Comment> comments) {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        return json;
+    }
```

Comment:
Is this method still used?

Reply:
It's not used but I thought it was a good idea to have around. I will probably implement it in my doGet or delete it.